### DISCUSSION TITLE
Calling a non-view function with `callStatic` using Wagmi

### DISCUSSION THREAD

**Discussion by akshatmittal** [wevm/wagmi#1571] [Q&A] [answered]
Basically title. Is there a way to use `wagmi`'s `useContractRead` style hooks to read data from a contract with callStatic? The function internally has state changes which is required for correct accounting but must be called with callStatic when looking to just get the data.

**Comment by tmm** [comment]
Check out [`useContract`](https://wagmi.sh/react/hooks/useContract). If you pass in a TypeScript [compatible ABI](https://wagmi.sh/react/typescript#contract-abis), you will get a fully type-safe Contract object.

**Reply by akshatmittal** [reply]
Yeah so I'm not just looking for a type-safe `ethers.Contract` object, but rather having all of the `useContractRead` abilities (such as `watch`) when calling a function as `callStatic`. Does that make sense?

**Reply by tmm** [reply]
Got it. Unfortunately this declarative behavior doesn't exist for `callStatic`.

**Reply by akshatmittal** [reply, marked as answer]
Yeah, I totally understand that. It would be pretty cool to have it in `wagmi`, consider it a feature request!

As of now, the workaround I've found is just to edit the `ABI` to set the `stateMutability` to `view` which just works with `useContractRead` haha.

**Reply by j5h46bd** [reply]
Thanks @akshatmittal, this is exactly what I needed.
